include/xtensor/xcontainer.hpp

```diff
@@ -1335,14 +1335,15 @@ namespace xt
     {
         if (m_shape.size() != shape.size() || !std::equal(std::begin(shape), std::end(shape), std::begin(m_shape)) || force)
         {
-            if (m_layout == layout_type::dynamic || m_layout == layout_type::any)
```

Comment:
note there is a static assert that containers can never be any

Reply:
even though I wonder if we should make vectors (1D arrays) have `layout_type::any` ...